packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.test.js

```diff
@@ -351,4 +351,48 @@ describe('useAutocomplete', () => {
       fireEvent.click(button);
     }).not.to.throw();
   });
+
+  it('should call getOptionLabel only with existing options on filtered options change', () => {
```

Comment:
Let's not test the code implementation.  Instead, we can verify that it does not throw an error when nested options are provided to `Autocomplete`. We should trigger the user interactions step by step and check that it does not crash in the end as provided in the issue.

1. The user opens the list and selects an option.
2. The user clears the selected option from input.
3. The user reopens the autocomplete.
4. It should not crash.

Also, please add this test in `Autocomplete` test file (`Autocomplete.test.js`). Let me know if you need any help.

Reply:
Thanks, I added the unit test for the `Autocomplete` component in `Autocomplete.test.js`.

I have also simplified and left the `useAutocomplete` test. It may be worth testing it on the hook level too, as in the future the `Autocomplete` component layer may have extra logic wrapping that behavior. If you prefer though, I could delete the part from the `useAutocomplete.test.js`.

Reply:
Please delete the test from the `useAutocomplete.test.js`. Any logic we have in future related to highlighting will be in the `useAutocomplete` hook only which will always be used in the `Autocomplete` component.